orion-ui/src/services/uiSettings.ts

```diff
@@ -14,8 +14,7 @@ export class UiSettings {
   public static settings: Settings | null = null
 
   private static promise: Promise<Settings> | null = null
-  private static readonly baseUrl = MODE() === 'development' ? 'http://127.0.0.1:4200' : window.location.origin
-
+  private static readonly baseUrl = MODE() === 'development' ? 'http://127.0.0.1:4200' : (BASE_URL() == null ? window.location.origin : BASE_URL())
```

Comment:
```suggestion
  private static readonly baseUrl = MODE() === 'development' ? 'http://127.0.0.1:4200' : (BASE_URL() ?? window.location.origin)
```
this does almost the same thing (except captures undefined as well) slightly more concisely

Reply:
Yes. Much cleaner.